### PR TITLE
Add storage settings with upload file size limit

### PR DESCRIPTION
**Pull request by thebengeu**
Screenshot 2022-02-22 at 15 57 37
Screenshot 2022-02-22 at 16 06 24

### CODE REVIEW

File: studio/lib/constants/infrastructure.ts [hunk 1]
```diff
@@ -132,3 +132,5 @@ export const PASSWORD_STRENGTH_PERCENTAGE = {
 }
 
 export const DEFAULT_FREE_PROJECTS_LIMIT = 2
+
+export const STORAGE_FILE_SIZE_LIMIT_MAX = 5_368_709_120
```
Comment (joshenlim) [inline studio/lib/constants/infrastructure.ts:136]: likewise from the API could we display this as a product (x * y * z) so its easier to see how many MB

**Comment by joshenlim** [comment]
just to clarify - so for free tier, their file upload size is fixed - users cant reduce it below 50mb?

**Comment by thebengeu** [comment]
> just to clarify - so for free tier, their file upload size is fixed - users cant reduce it below 50mb?

As far as I know, no one has ever asked us to reduce their upload file size limit, so I don't think we should complicate things unless that comes up as a common need. Also, if we start off by disabling updates entirely, even reductions, we reserve the option to ask users to upgrade plan if they want to do any customisation of their upload file size limit.

**Comment by joshenlim** [comment]
gotcha yup! sounds good!

**Comment by joshenlim** [comment]
i'm presuming that the API PR has to go in first before studio right?

**Comment by joshenlim** [comment]
actually on that note - i'm guessing the phrasing on the storage settings page for free tier needs to clarify this? i'll push an update

**Comment by thebengeu** [comment]
> i'm presuming that the API PR has to go in first before studio right?

Yes, the API PR is now merged.

> actually on that note - i'm guessing the phrasing on the storage settings page for free tier needs to clarify this?
> 
> maybe
> "Free Plan has a fixed upload file size limit of 50 MB."
> "Please upgrade to Pro plan for a configurable upload file size limit of up to 5 GB."
> 
> wdyt?

That's great, thank you!

**Review by joshenlim — APPROVED** [review]

**Pull request merged**